packages/zero-client/src/client/query-manager.test.ts

```diff
@@ -66,10 +66,52 @@ test('add', () => {
     },
   ]);
 
-  queryManager.add(ast, 'forever');
+  queryManager.addLegacy(ast, 'forever');
   expect(send).toBeCalledTimes(1);
 });
 
+test('add and remove a custom query', () => {
+  const send = vi.fn<(arg: ChangeDesiredQueriesMessage) => void>();
+  const maxRecentQueriesSize = 0;
+  const mutationTracker = new MutationTracker(lc);
+  const queryManager = new QueryManager(
+    mutationTracker,
+    'client1',
+    schema.tables,
+    send,
+    () => () => {},
+    maxRecentQueriesSize,
+  );
+  const rm1 = queryManager.addCustom('customQuery', [1], 'forever');
+  expect(send).toBeCalledTimes(1);
+  expect(send).toBeCalledWith([
+    'changeDesiredQueries',
+    {
+      desiredQueriesPatch: [
+        {
+          op: 'put',
+          hash: '2tts1u0ojmujd',
+          name: 'customQuery',
+          args: [1],
+          ttl: -1,
+        },
+      ],
+    },
+  ]);
+
+  const rm2 = queryManager.addCustom('customQuery', [1], 'forever');
+  expect(send).toBeCalledTimes(1);
+
+  rm2();
+  const rm3 = queryManager.addCustom('customQuery', [1], 'forever');
+  expect(send).toBeCalledTimes(1);
+  rm1();
+  rm3();
+  queryManager.addCustom('customQuery', [1], 'forever');
+  // once for del, another for put
```

Comment:
Might be a bit better to assert the send for del before calling addCustom again and asserting the contents of what was sent.